### PR TITLE
Extract CConnman::RelayTransaction

### PR DESCRIPTION
**Pull request by Empact**
* Implement `RelayTransaction` on `CConnman` only, rather than chain interface, `BroadcastTransaction`, and locally in net processing.
* Split node `PushInventory` into `PushBlockInventory` and `PushTransactionInventory`, as at every call-site the block/tx distinction is already made.

Extracted from #15253

### CODE REVIEW

**Comment by promag** [comment]
utACK d48c4bf, nice and simple refactor.

**Comment by gmaxwell** [comment]
I don't see what value this adds, particularly doing things like creating functions to encapsulate single lines of code.

**Comment by Empact** [comment]
As noted I wrote this to enable the logging introduced here: https://github.com/bitcoin/bitcoin/commit/389709e02c165ef06db35986e1619c16824f880f

In terms of justifying it independently, in the first case, having a single implementation protects against unintentional divergence going forward (the DRY principle), also simplifies by removing a layer of indirection relative to CConnman.

Dropped the second commit as I found no satisfying independent justification for it.

**Comment by Empact** [comment]
This was originally requested by @MarcoFalke here - that is, unless I misinterpreted him: https://github.com/bitcoin/bitcoin/pull/15253#issuecomment-490534814

**Comment by Empact** [comment]
On second look, consider #15945. `PushInventory` currently silently ignores unrecognized INV types, which is a potential point of confusion for ineffective callers. This narrows the `Push*Inventory` calls to those which can and will be handled. Restored the initial version.

File: src/net.h [hunk 1]
```diff
@@ -839,17 +842,11 @@ class CNode
...
 
-    void PushInventory(const CInv& inv)
-    {
-        LOCK(cs_inventory);
-        if (inv.type == MSG_TX) {
-            if (!filterInventoryKnown.contains(inv.hash)) {
-                setInventoryTxToSend.insert(inv.hash);
-            }
-        } else if (inv.type == MSG_BLOCK) {
-            vInventoryBlockToSend.push_back(inv.hash);
```
Comment (MarcoFalke) [inline src/net.h]: Could keep the implementation inlined in the header to keep the diff smaller?

**Comment by sdaftuar** [comment]
IMO it makes more sense to refactor transaction/block relay logic out of net.cpp and into net_processing.cpp.  So further consolidating this in connman seems like a small step in the wrong direction, and not worth the friction of changing around how this logic works.

**Comment by Empact** [comment]
@sdaftuar Not sure what you have in mind, but it's now in a separate file rather than on `CConnman`.

**Comment by sdaftuar** [comment]
IMO it doesn't seem like this accomplishes anything, and it conflicts with some other PRs, so I would tend to NACK.

**Comment by MarcoFalke** [comment]
Seems too controversial for a simple change such as this. Going to close this for now.

**Pull request closed without merging**